src/ext.rs

```diff
@@ -79,9 +91,11 @@ where I: Traversal<T> {
     }
 }
 
-impl<T, I, F> Traversal<T> for SkipWhile<I, F>
-where I: Traversal<T>, F: FnMut(&T) -> bool {
-    fn foreach<F1: FnMut(T) -> bool>(self, mut f: F1) {
+impl<T, I, F> Traversal for SkipWhile<I, F>
+where I: Traversal<Item=T>, F: FnMut(&T) -> bool {
```

Comment:
I usually leave stuff like this for short lines/bounds and split it up when things get really complex.